File: Stack/2_push_pop.js
class Stack
{
    constructor(size)
    {
        this.size = size;
        this.top  = -1;
        this.data = {};
    }

    
    push(element)
    {
        if(this.isFull())
        {
            console.log("Can not push to already full stack");
            return;
        }

        this.top++;
        this.data[this.top] = element;
    }

    pop()
    {
        if(this.isEmpty())
        {
            console.log("Can not pop from an empty stack");
            return -1;
        }
        // top--;
        return this.data[this.top--];
    }

    //if top is less than 0 it means it is not poiting to any element of stack 
    //thus stack is empty
    isEmpty()
    {
        if (this.top < 0)
            return true;
        return false;
    }

    isFull()
    {
         if (this.top == this.size-1)
            return true;
        return false;

    }

    display()
    {
        for(let i=0;i<=this.top;i++)
        {
            console.log(this.data[i]);
        }
    }

}

s = new Stack(5);
s.push(1);
s.push(4);
s.push(3);
s.display();
console.log('popped: ',s.pop());
s.pop();
s.pop();
s.display();

File: Stack/2_push_pop.py
class Stack:

    def __init__(self,size):
        self.size = size
        self.top  = -1
        self.data = [None]*size
        #data list of given size
    
    def push(self,element):
        if(self.isFull()): 
            print("Can not push to already full stack")
            return

        self.top += 1
        self.data[self.top] = element
    

    def pop(self):
        if(self.isEmpty()):
            print("Can not pop from an empty stack")
            return -1
        
        self.top -= 1
        return self.data[self.top+1]
    

    #if top is less than 0 it means it is not poiting to any element of stack 
    #thus stack is empty
    def isEmpty(self):
        if (self.top < 0):
            return True
        return False
    

    def isFull(self):
        if (self.top == self.size-1):
            return True
        return False

    def display(self):
    
        print("[",end=" ")
        # for(int i=0;i<=top;i++)
        for i in range(0,self.top+1):
            # print(self.data[i],", ")
            print(self.data[i],end=", ")
        print("]")
        
    
s = Stack(5)
s.push(1)
s.push(4)
s.push(3)
s.display()
print('popped: ',s.pop())
s.pop()
s.pop()
s.display()